### PR TITLE
Drafts download and deployment docs for ELSER

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds docs on how to download and deploy ELSER.

### Preview

[ELSER](https://stack-docs_2371.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-nlp-elser.html)

### CODE REVIEW

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc [hunk 1]
```diff
@@ -7,4 +7,116 @@
...
+After you downloaded ELSER, you can use {kib} to view and manage its deployment 
+across your cluster under **{ml-app}** > **Trained Models**.
+
+1. Start the deployment by clicking the Start deployment button.
+2. Provide a deployment ID, select priority, and set the number of allocations 
+and threads per allocation values.
++
+--
+[role="screenshot"]
+image::images/ml-nlp-elser-deployment.png[alt="Deploying ELSER",align="center"]
```
Comment (szabosteve) [inline docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:98]:
```suggestion
image::images/ml-nlp-deploy-elser.png[alt="Deploying ELSER",align="center"]
```

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc [hunk 2]
```diff
@@ -7,4 +7,116 @@
 :keywords: {ml-init}, {stack}, {nlp}, ELSER
 :description: ELSER is a learned sparse ranking model trained by Elastic.
 
-[TBD]
\ No newline at end of file
+Elastic Learned Sparse Encoder - or ELSER - is a large language model (LLM)
```
Comment (droberts195) [inline docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:10]: Please add the tech preview banner to this page.

**Review by qherreros — COMMENTED** [review]
Very nice work! I made a small suggestion on ELSER.

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc [hunk 3]
```diff
@@ -7,4 +7,108 @@
...
+To use ELSER, you must have the {subscriptions}[appropriate subscription] level 
+or the trial period activated.
+
+
+[discrete]
+[[download-elser]]
+== Download ELSER
+
+You can download ELSER either from the Trained Models page in {kib} or by using 
+the Dev Console.
```
Comment (droberts195) [inline docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:39]: Also from Enterprise Search - see https://github.com/elastic/kibana/pull/156080.

I am not sure how heavily we should push people towards the Enterprise Search app in the ML docs, but Enterprise Search is where we're planning to make the semantic search experience as friendly as possible, so we should certainly mention it here.

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc [hunk 4]
```diff
@@ -7,4 +7,108 @@
 :keywords: {ml-init}, {stack}, {nlp}, ELSER
 :description: ELSER is a learned sparse ranking model trained by Elastic.
 
-[TBD]
\ No newline at end of file
+experimental[]
+
+Elastic Learned Sparse Encoder - or ELSER - is a text expansion model that uses 
+sparse vector representation and trained by Elastic . As a retrieval model,
```
Comment (davidkyle) [inline docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:13]:
```suggestion
Elastic Learned Sparse Encoder - or ELSER - is a representational model trained by Elastic that creates a
sparse vector representation  . As a retrieval model, 
```

Text expansion the _user friendly_ name we have chosen for the search query but it isn't a common term that people will recognise. I like @qherreros description of expanding text, maybe use the 'text expansion' after that

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc (hunk 3 above)
Comment (szabosteve) [reply docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:39]: I'm adding a section that covers this method, too.

**Comment by szabosteve** [comment]
@qherreros @droberts195 @davidkyle I updated the PR with your suggestions. In [8ac8269](https://github.com/elastic/stack-docs/pull/2371/commits/8ac82693ef28ea69b3e580b1e347ea7841794e01), I added a new section that covers the steps of deploying the model from the Enterprise Search app.

**Review by qherreros — APPROVED** [review]
LGTM !

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc [hunk 5]
```diff
@@ -9,6 +9,125 @@
...
+To use ELSER, you must have the {subscriptions}[appropriate subscription] level 
+or the trial period activated.
+
+
+[discrete]
+[[download-elser]]
+== Download ELSER
+
+You can download ELSER either from the **{ml-app}** > **Trained Models** page or 
+by using the Dev Console.
```
Comment (droberts195) [inline docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:40]:
```suggestion
You can download ELSER either from the **{ml-app}** > **Trained Models** page, 
by using the Dev Console, or using {ents}.
```

Comment (szabosteve) [reply docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:40]: @droberts195 If I understand [this PR](https://github.com/elastic/kibana/pull/156080) correctly, you need to have ELSER downloaded in order to deploy the model through the Enterprise Search application. This is the reason why I didn't mention Enterprise Search here, only in the section about deploying the model.

Comment (szabosteve) [reply docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc]: Okay, I modified in [94d47dd](https://github.com/elastic/stack-docs/pull/2371/commits/94d47dda64547413fa17e280efaa30988499e8cc).

Comment (demjened) [reply docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc]: > it's possible to both download and deploy ELSER just using the Enterprise Search app right?

Yes, correct.

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc [hunk 6]
```diff
@@ -9,19 +9,137 @@
 
 experimental[]
 
+
+Elastic Learned Sparse Encoder - or ELSER - is a representational model trained
```
Comment (davidkyle) [inline docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:13]:
```suggestion
Elastic Learned Sparse EncodeR - or ELSER - is a representational model trained 
```

For the R in ELSER. Is that correct @qherreros

**Review by davidkyle — APPROVED** [review]
LGTM

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc (hunk 6 above)
Comment (qherreros) [reply docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:13]: Yes, true!

File: docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc [hunk 7]
```diff
@@ -113,3 +231,11 @@ xpack.ml.model_repository: file://${path.home}/config/models/`
 . Repeat step 2 and step 3 on all master-eligible nodes.
 . {ref}/restart-cluster.html#restart-cluster-rolling[Restart] the 
 master-eligible nodes one by one.
+
+
+[discrete]
+[[further-readings]]
+== Further readings
```
Comment (droberts195) [inline docs/en/stack/ml/nlp/ml-nlp-elser.asciidoc:238]:
```suggestion
== Further reading
```

**Review by droberts195 — APPROVED** [review]
LGTM

**Pull request merged**